core/core.go

```diff
@@ -18,12 +18,14 @@ import (
 type Core struct {
 	registry map[msg.ChainId]Chain
 	route    *router.Router
+	log      log15.Logger
 }
 
 func NewCore() *Core {
 	return &Core{
 		registry: make(map[msg.ChainId]Chain),
-		route:    router.NewRouter(),
+		route:    router.NewRouter(log15.New("router")),
+		log:      log15.New("core"),
```

Comment:
these will get prefixed with module (`module=router`) since you only provide 1 arg, let rather use `system` by specifying that as the first arg for both loggers.